apps/settings/src/components/UserList/UserRow.vue

```diff
@@ -99,151 +100,146 @@
-				ref="mailAddress"
+			<NcTextField :id="'mailAddress'+user.id+rand"
+				:show-trailing-button="true"
+				class="user-row-text-field"
+				:class="{'icon-loading-small': loading.mailAddress}"
 				:disabled="loading.mailAddress||loading.all"
-				:placeholder="t('settings', 'Add new email address')"
-				:value="user.email"
+				trailing-button-icon="arrowRight"
+				:value.sync="editedMail"
 				autocapitalize="off"
 				autocomplete="new-password"
 				autocorrect="off"
 				spellcheck="false"
-				type="email">
-			<input class="icon-confirm" type="submit" value="">
-		</form>
+				type="email"
+				@trailing-button-click="updateEmail">
+				<template #trailing-button-icon>
+					<ArrowRight :size="20" />
+				</template>
+			</NcTextField>
+		</div>
 		<div :class="{'icon-loading-small': loading.groups}" class="groups">
 			<label class="hidden-visually" :for="'groups'+user.id+rand">{{ t('settings', 'Add user to group') }}</label>
-			<NcMultiselect :id="'groups'+user.id+rand"
+			<NcSelect :input-id="'groups'+user.id+rand"
 				:close-on-select="false"
 				:disabled="loading.groups||loading.all"
-				:limit="2"
 				:multiple="true"
 				:options="availableGroups"
 				:placeholder="t('settings', 'Add user to group')"
-				:tag-width="60"
 				:taggable="settings.isAdmin"
 				:value="userGroups"
-				class="multiselect-vue"
+				class="select-vue"
 				label="name"
-				tag-placeholder="create"
-				track-by="id"
-				@remove="removeUserGroup"
-				@select="addUserGroup"
-				@tag="createGroup">
-				<span slot="noResult">{{ t('settings', 'No results') }}</span>
-			</NcMultiselect>
+				:no-wrap="true"
+				:selectable="() => userGroups.length < 2"
+				:create-option="(value) => ({ name: value, isCreating: true })"
+				@option:created="createGroup"
+				@option:selected="addUserGroup($event.at(-1))"
+				@option:deselected="removeUserGroup" />
 		</div>
 		<div v-if="subAdminsGroups.length>0 && settings.isAdmin"
 			:class="{'icon-loading-small': loading.subadmins}"
 			class="subadmins">
 			<label class="hidden-visually" :for="'subadmins'+user.id+rand">{{ t('settings', 'Set user as admin for') }}</label>
-			<NcMultiselect :id="'subadmins'+user.id+rand"
+			<NcSelect :id="'subadmins'+user.id+rand"
 				:close-on-select="false"
 				:disabled="loading.subadmins||loading.all"
-				:limit="2"
+				label="name"
 				:multiple="true"
+				:no-wrap="true"
+				:selectable="() => userSubAdminsGroups.length < 2"
 				:options="subAdminsGroups"
 				:placeholder="t('settings', 'Set user as admin for')"
-				:tag-width="60"
 				:value="userSubAdminsGroups"
-				class="multiselect-vue"
-				label="name"
-				track-by="id"
-				@remove="removeUserSubAdmin"
-				@select="addUserSubAdmin">
-				<span slot="noResult">{{ t('settings', 'No results') }}</span>
-			</NcMultiselect>
+				class="select-vue"
+				@option:deselected="removeUserSubAdmin"
+				@option:selected="addUserSubAdmin($event.at(-1))" />
```

Comment:
```suggestion
				@option:selected="options => addUserSubAdmin(options.at(-1))" />
```
Seems a bit obtuse, let's make it more explicit